src/main.rs

```diff
@@ -15,6 +17,32 @@ use termion::screen::IntoAlternateScreen;
  * @return Result
  */
 fn main() -> Result<(), Box<dyn Error>> {
+
+	let raw_output = io::stdout().into_raw_mode()?;
+	raw_output.suspend_raw_mode()?;
+
+	let panic_hook = panic::take_hook();
+
+	panic::set_hook(Box::new (move |panic| {
+		let panic_cleanup = || -> Result<(), Box<dyn Error>> {
+			let mut output = io::stdout();
+
+			write!(
+				output,
+				"{}{}{}",
+				termion::clear::All,
+				termion::screen::ToMainScreen,
+				termion::cursor::Show
+			)?;
+			raw_output.suspend_raw_mode()?;
+			output.flush()?;
+			Ok(())
+		};
+
+		panic_cleanup().expect("Failed to cleanup after panic");
+		panic_hook(panic);
+	}));
```

Comment:
Maybe it would be better to wrap this in a function called e.g. `setup_panic_hook` and move it to `util`? Then we can call that function after line 55 (just before creating the `backend`) which is I think more appropriate.

Reply:
@orhun please check the latest update and let me know if implementation is correct. As for the functionality it's working same as before.